yoyodyne/sizing.py

```diff
@@ -87,53 +54,68 @@ def find_batch_size(
     model: models.BaseEncoderDecoder,
     datamodule: data.DataModule,
     *,
-    max_trials: int = defaults.FIND_BATCH_SIZE_MAX_TRIALS,
-    mode: str = defaults.FIND_BATCH_SIZE_MODE,
     steps_per_trial: int = defaults.FIND_BATCH_SIZE_STEPS_PER_TRIAL,
 ) -> None:
     """Computes maximum or optimal batch size.
 
     This sets the batch size on the datamodule, and if necessary, the
     gradient accumulation steps on the trainer.
 
+    With the max method, batch size is doubled until a memory error, thus
+    finding the largest batch size that fits in memory.
+
+    With the optimal method, batch size is doubled until a memory error or
+    once we confirm that the desired batch size fits in memory (whichever
+    comes first).
+
     Args:
         method (str): one of "max" (find the maximum batch size) or "opt" (find
             the "optimal" batch size, using the gradient accumulation trick if
             necessary.)
         trainer (pl.Trainer).
         model (models.BaseEncoderDecoder).
         datamodule (data.DataModule).
-        max_trials (int, optional): maximum number of increases in batch size
-            before terminating.
-        mode (str, optional): one of "binsearch" or "power".
```

Comment:
Why are we removing this?

Reply:
So I have tested this relativley extensively and you absolutely do not want to use binsearch. It takes orders of magnitude more time and it then immediately crashes with an OOM anyways because our batch lengths vary.